resolver.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prototransform
+
+import (
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+// Resolver is used to resolve symbol names and numbers into schema definitions.
+type Resolver interface {
+	protoregistry.ExtensionTypeResolver
+	protoregistry.MessageTypeResolver
+
+	// FindEnumByName looks up an enum by its full name.
+	// E.g., "google.protobuf.Field.Kind".
+	//
+	// This returns (nil, NotFound) if not found.
+	FindEnumByName(enum protoreflect.FullName) (protoreflect.EnumType, error)
+}
+
+// resolver is the interface used by and provided by a *SchemaWatcher, which is
+// intentionally broader than the above interface.
+type resolver interface {
```

Comment:
If it is internal why not just use a concrete type?

Reply:
Doh, good point. This interface is not necesary.

Reply:
Done